fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -198,7 +188,11 @@ private static void processViews(QueryStatement queryStatement, List<BaseTableIn
         }
         Set<ViewRelation> viewRelationSet = Sets.newHashSet(viewRelations);
         for (ViewRelation viewRelation : viewRelationSet) {
+            // base tables of view
             processBaseTables(viewRelation.getQueryStatement(), baseTableInfos);
+
+            // view itself is considered as base-table
+            baseTableInfos.add(BaseTableInfo.fromTableName(viewRelation.getName(), viewRelation.getView()));
```

Comment:
Why add view as the base table? To keep the relation between MV and view?

Reply:
To make the MV be aware of view changing, specifically you alter the view, the MV will be inactivated